File: backend/app/models/appointment.rb
class Appointment < ApplicationRecord
  belongs_to :user, optional: true

  # Validations
  validates :name, presence: true
  validates :email, presence: true, format: { with: URI::MailTo::EMAIL_REGEXP }
  validates :phone, presence: true
  validates :preferred_language, presence: true
  validates :status, presence: true

  # Status enum
  enum status: {
    pending: 'pending',
    confirmed: 'confirmed',
    completed: 'completed',
    cancelled: 'cancelled'
  }

  # Default status
  after_initialize :set_default_status, if: :new_record?

  private

  def set_default_status
    self.status ||= 'pending'
  end
end

File: frontend/src/types/appointment.ts
export interface AppointmentFormData {
  name: string;
  email: string;
  phone: string;
  message?: string;
  preferred_language: string;
  service_id?: number;
  preferred_date?: string | null;
  child_name?: string;
  child_age?: string;
}

export interface Appointment extends AppointmentFormData {
  id: number;
  status: 'pending' | 'confirmed' | 'completed' | 'cancelled';
  user_id?: number;
  user_name?: string;
  service_name?: string | null;
  service_price?: number | null;
  created_at: string;
  updated_at: string;
}

export interface AppointmentResponse {
  success: boolean;
  message: string;
  data: { id: number };
}
